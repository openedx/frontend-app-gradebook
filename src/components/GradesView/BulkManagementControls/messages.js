import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  downloadGradesBtn: {
    id: 'gradebook.GradesView.BulkManagementControls.bulkManagementLabel',
    defaultMessage: 'Download Grades',
    description: 'A labeled button that allows an admin user to download course grades all at once (in bulk).',
  },
});

export default messages;
