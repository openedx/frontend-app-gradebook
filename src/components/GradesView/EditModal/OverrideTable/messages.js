import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  adjustedGradeHeader: {
    id: 'gradebook.GradesView.EditModal.Overrides.adjustedGradeHeader',
    defaultMessage: 'Adjusted grade',
    description: 'Edit Modal Override Table Adjusted grade column header',
  },
  dateHeader: {
    id: 'gradebook.GradesView.EditModal.Overrides.dateHeader',
    defaultMessage: 'Date',
    description: 'Edit Modal Override Table Date column header',
  },
  graderHeader: {
    id: 'gradebook.GradesView.EditModal.Overrides.graderHeader',
    defaultMessage: 'Grader',
    description: 'Edit Modal Override Table Grader column header',
  },
  reasonHeader: {
    id: 'gradebook.GradesView.EditModal.Overrides.reasonHeader',
    defaultMessage: 'Reason',
    description: 'Edit Modal Override Table Reason column header',
  },
});

export default messages;
