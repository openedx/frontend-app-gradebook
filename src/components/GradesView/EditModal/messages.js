import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  assignmentHeader: {
    id: 'gradebook.GradesView.EditModal.headers.assignment',
    defaultMessage: 'Assignment',
    description: 'Edit Modal Assignment header',
  },
  currentGradeHeader: {
    id: 'gradebook.GradesView.EditModal.headers.currentGrade',
    defaultMessage: 'Current Grade',
    description: 'Edit Modal Current Grade header',
  },
  originalGradeHeader: {
    id: 'gradebook.GradesView.EditModal.headers.originalGrade',
    defaultMessage: 'Original Grade',
    description: 'Edit Modal Original Grade header',
  },
  studentHeader: {
    id: 'gradebook.GradesView.EditModal.headers.student',
    defaultMessage: 'Student',
    description: 'Edit Modal Student header',
  },
  title: {
    id: 'gradebook.GradesView.EditModal.title',
    defaultMessage: 'Edit Grades',
    description: 'Edit Modal title',
  },
  closeText: {
    id: 'gradebook.GradesView.EditModal.closeText',
    defaultMessage: 'Cancel',
    description: 'Edit Modal close button text',
  },
  visibility: {
    id: 'gradebook.GradesView.EditModal.contactSupport',
    defaultMessage: 'Showing most recent actions (max 5). To see more, please contact support',
    description: 'Edit Modal visibility hint message',
  },
  saveVisibility: {
    id: 'gradebook.GradesView.EditModal.saveVisibility',
    defaultMessage: 'Note: Once you save, your changes will be visible to students.',
    description: 'Edit Modal saved changes effect hint',
  },
  saveGrade: {
    id: 'gradebook.GradesView.EditModal.saveGrade',
    defaultMessage: 'Save Grades',
    description: 'Edit Modal Save button label',
  },
});

export default messages;
