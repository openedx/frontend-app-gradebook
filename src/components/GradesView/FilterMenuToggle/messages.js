import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  editFilters: {
    id: 'gradebook.GradesView.editFilterLabel',
    defaultMessage: 'Edit Filters',
    description: 'A labeled button in the Grades tab that opens/closes the Filters tab, allowing the grades to be filtered',
  },
});

export default messages;
