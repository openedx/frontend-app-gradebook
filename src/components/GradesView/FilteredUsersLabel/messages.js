import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  visibilityLabel: {
    id: 'gradebook.GradesTab.usersVisibilityLabel',
    defaultMessage: 'Showing {filteredUsers} of {totalUsers} total learners',
    description: 'Users visibility label',
  },
});

export default messages;
