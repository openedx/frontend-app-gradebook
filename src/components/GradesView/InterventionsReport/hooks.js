import { actions, selectors } from 'data/redux/hooks';

const useInterventionsReportData = () => {
  const interventionExportUrl = selectors.root.useInterventionExportUrl();
  const showBulkManagement = selectors.root.useShowBulkManagement();
  const downloadInterventionReport = actions.grades.useDownloadInterventionReport();

  const handleClick = () => {
    downloadInterventionReport();
    window.location.assign(interventionExportUrl);
  };

  return {
    show: showBulkManagement,
    handleClick,
  };
};

export default useInterventionsReportData;
