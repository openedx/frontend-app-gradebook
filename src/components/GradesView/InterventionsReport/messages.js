import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  title: {
    id: 'gradebook.GradesView.InterventionsReport.title',
    defaultMessage: 'Interventions Report',
    description: 'The title for the Intervention report subsection',
  },
  description: {
    id: 'gradebook.GradesView.InterventionsReport.description',
    defaultMessage: 'Need to find students who may be falling behind?  Download the interventions report to obtain engagement metrics such as section attempts and visits.',
    description: 'The description for the Intervention report subsection',
  },
  downloadBtn: {
    id: 'gradebook.GradesView.InterventionsReport.downloadBtn',
    defaultMessage: 'Download Interventions',
    description: 'The labeled button to download the Intervention report from the Grades View',
  },
});

export default messages;
