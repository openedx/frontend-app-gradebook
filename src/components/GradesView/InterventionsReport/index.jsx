import React from 'react';

import { useIntl } from '@edx/frontend-platform/i18n';

import NetworkButton from 'components/NetworkButton';

import messages from './messages';
import useInterventionsReportData from './hooks';

/**
 * <InterventionsReport />
 * Provides download buttons for Bulk Management and Intervention reports, only if
 * showBulkManagement is set in redus.
 */
export const InterventionsReport = () => {
  const { show, handleClick } = useInterventionsReportData();
  const { formatMessage } = useIntl();

  if (!show) {
    return null;
  }

  return (
    <div>
      <h4 className="mt-0">
        {formatMessage(messages.title)}
      </h4>
      <div
        className="d-flex justify-content-between align-items-center"
      >
        <div className="intervention-report-description">
          {formatMessage(messages.description)}
        </div>
        <NetworkButton
          label={messages.downloadBtn}
          onClick={handleClick}
        />
      </div>
    </div>
  );
};

export default InterventionsReport;
