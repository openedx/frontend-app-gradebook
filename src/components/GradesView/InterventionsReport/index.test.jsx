import React from 'react';
import { shallow } from 'enzyme';

import { useIntl } from '@edx/frontend-platform/i18n';

import NetworkButton from 'components/NetworkButton';

import messages from './messages';
import useInterventionsReportData from './hooks';
import InterventionsReport from '.';

jest.mock('components/NetworkButton', () => 'NetworkButton');
jest.mock('./hooks', () => jest.fn());

const hookProps = { show: true, handleClick: jest.fn() };
useInterventionsReportData.mockReturnValue(hookProps);

let el;
describe('InterventionsReport component', () => {
  beforeEach(() => {
    el = shallow(<InterventionsReport />);
  });
  describe('behavior', () => {
    it('initializes hooks', () => {
      expect(useInterventionsReportData).toHaveBeenCalledWith();
      expect(useIntl).toHaveBeenCalledWith();
    });
  });
  describe('output', () => {
    it('does now render if show is false', () => {
      useInterventionsReportData.mockReturnValueOnce({ ...hookProps, show: false });
      el = shallow(<InterventionsReport />);
      expect(el.isEmptyRender()).toEqual(true);
    });
    test('snapshot', () => {
      expect(el).toMatchSnapshot();
      const btnProps = el.find(NetworkButton).props();
      expect(btnProps.label).toEqual(messages.downloadBtn);
      expect(btnProps.onClick).toEqual(hookProps.handleClick);
    });
  });
});
