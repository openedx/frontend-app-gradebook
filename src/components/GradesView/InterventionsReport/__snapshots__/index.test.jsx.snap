// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterventionsReport component output snapshot 1`] = `
<div>
  <h4
    className="mt-0"
  >
    Interventions Report
  </h4>
  <div
    className="d-flex justify-content-between align-items-center"
  >
    <div
      className="intervention-report-description"
    >
      Need to find students who may be falling behind?  Download the interventions report to obtain engagement metrics such as section attempts and visits.
    </div>
    <NetworkButton
      label={
        Object {
          "defaultMessage": "Download Interventions",
          "description": "The labeled button to download the Intervention report from the Grades View",
          "id": "gradebook.GradesView.InterventionsReport.downloadBtn",
        }
      }
      onClick={[MockFunction]}
    />
  </div>
</div>
`;
